File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/database/DatabaseTestResult.java
package com.mieai.qqbot.admin.database;

import java.time.Instant;

public record DatabaseTestResult(
        boolean success,
        DatabaseType type,
        String databaseProduct,
        String databaseVersion,
        long latencyMs,
        boolean readVerified,
        boolean writeVerified,
        DatabaseSchemaState schemaState,
        String schemaVersion,
        boolean initialized,
        boolean adminSeeded,
        long botCount,
        Instant testedAt) {
}

File: qqbot-module-database-support/src/main/kotlin/com/mieai/qqbot/admin/database/DatabaseTestResult.kt
package com.mieai.qqbot.admin.database

import java.time.Instant

data class DatabaseTestResult(
    val success: Boolean,
    val type: DatabaseType,
    val databaseProduct: String?,
    val databaseVersion: String?,
    val latencyMs: Long,
    val readVerified: Boolean,
    val writeVerified: Boolean,
    val schemaState: DatabaseSchemaState,
    val schemaVersion: String?,
    val initialized: Boolean,
    val adminSeeded: Boolean,
    val botCount: Long,
    val testedAt: Instant,
)
